solarforecastarbiter/reference_forecasts/main.py

```diff
@@ -198,125 +206,154 @@ def load_data(observation, data_start, data_end):
-    forecast_end = forecast_start + forecast.run_length
-    if 'instant' in forecast.interval_label:
-        forecast_end -= pd.Timedelta('1s')
-    return forecast_start, forecast_end
-
-
-def _is_intraday(forecast):
-    """Is the forecast intraday?"""
-    # intra day persistence and "day ahead" persistence require
-    # fairly different parameters.
-    # is this a sufficiently robust way to distinguish?
-    return forecast.run_length < pd.Timedelta('1d')
-
-
-def _check_midnight_to_midnight(forecast_start, forecast_end):
-    if (forecast_start.round('1d') != forecast_start or
-            forecast_end - forecast_start > pd.Timedelta('1d')):
-        raise ValueError(
-            'Day ahead persistence requires midnight to midnight periods')
-
-
-def _intraday_start_end(observation, forecast, run_time):
-    # time window over which observation data will be used to create
-    # persistence forecast.
-    if (observation.interval_length > forecast.run_length or
-            observation.interval_length > pd.Timedelta('1h')):
-        raise ValueError(
-            'Intraday persistence requires observation.interval_length '
-            '<= forecast.run_length and observation.interval_length <= 1h')
-    # no longer than 1 hour
-    window = min(forecast.run_length, pd.Timedelta('1hr'))
-    data_end = run_time
-    data_start = data_end - window
-    return data_start, data_end
-
-
-def _dayahead_start_end(run_time):
-    # day ahead persistence: tomorrow's forecast is equal to yesterday's
-    # observations. So, forecast always uses obs > 24 hr old at each valid
-    # time. Smarter approach might be to use today's observations up
-    # until issue_time, and use yesterday's observations for issue_time
-    # until end of day. So, forecast *never* uses obs > 24 hr old at each
-    # valid time. Arguably too much for a reference forecast.
-    data_end = run_time.floor('1d')
-    data_start = data_end - pd.Timedelta('1d')
-    return data_start, data_end
-
-
-def _adjust_for_instant_obs(data_start, data_end, observation, forecast):
-    # instantaneous observations require care.
-    # persistence models return forecasts with same closure as obs
-    if 'instant' in forecast.interval_label:
-        if forecast.interval_length != observation.interval_length:
-            raise ValueError('Instantaneous forecast requires instantaneous '
-                             'observation with identical interval length.')
+    df_vals = []
+    for fx in forecasts:
+        if fx.extra_parameters == '':
+            continue
+        try:
+            extra_parameters = json.loads(fx.extra_parameters)
+        except json.JSONDecodeError:
+            logger.warning(
+                'Failed to decode extra_parameters for %s: %s as JSON',
+                fx.name, fx.forecast_id)
+            continue
+
+        try:
+            model = extra_parameters['model']
+        except KeyError:
```

Comment:
if using the explicit indication of a reference forecast above, then this `except` block can probably be simplified.